x-pack/lib/monitoring/inputs/metrics/stats_event_factory.rb

```diff
@@ -7,14 +7,27 @@ class StatsEventFactory
     include ::LogStash::Util::Loggable
     require 'logstash/config/pipelines_info'
 
+    def self.format_cluster_uuid(stats)
+      result = stats.extract_metrics([:stats, :pipelines, :main, :config], :cluster_uuids)
+      if result && !result[:cluster_uuids].empty?
+        result[:cluster_uuids][0]
+      else
+        if LogStash::SETTINGS.set?("monitoring.cluster_uuid")
+          LogStash::SETTINGS.get("monitoring.cluster_uuid")
+        else
+          "<UNDEFINED cluster>"
```

Comment:
I believe this needs to be blank, rather than `<UNDEFINED cluster>` - if the `cluster_uuid` field is blank, then the stack monitoring page in Kibana will show a page with a list of clusters to select from, including a "standalone" cluster. The monitoring entries for a Logstash instance where the cluster_uuid cannot be found are located on the "standalone" cluster page.

I think we should also log here.